File: python/arrays/1726-tuple-with-same-product.py
from typing import List

def tupleSameProduct(nums: List[int]) -> int:
    distinct = 0
    prods = {}
    for i in range(len(nums) - 1):
        for j in range(i + 1, len(nums)):
            if nums[i] != nums[j]:
                prod = nums[i]*nums[j]
                prods[prod] = prods.get(prod, 0) + 1

    print(prods)

    for idx in prods:
        distinct += prods[idx] * (prods[idx] - 1) // 2

    return distinct * 8

File: js/arrays/1726-tuple-with-same-product.js
var tupleSameProduct = function(nums) {
    prods = {}
    for (var i = 0; i < nums.length; i++) {
        for (var j = i + 1; j < nums.length; j++) {
            key = nums[i] * nums[j];
            if (prods[key] == undefined) prods[key] = 1;
            else prods[key] += 1;
        }
    }
    
    var total = 0;
    for (const [key, value] of Object.entries(prods)) {
        if (value > 1) total += (value * (value - 1)) / 2;
    }
    
    return total * 8;
};